docs/runtime/types/module-struct.md

```diff
@@ -0,0 +1,241 @@
+
+## The Definition of  Module Struct
+
+Here is template runtime module provided by the node-template, it contains following code along with `decl_module!` macro:
+
+```rust
+decl_module! {
+    /// The module declaration.
+    pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+        // Initializing events
+        fn deposit_event<T>() = default;
+        
+        // Just a dummy entry point.
+        pub fn do_something(origin, something: u32) -> Result {
+            let who = ensure_signed(origin)?; 
+            <Something<T>>::put(something);
+            Self::deposit_event(RawEvent::SomethingStored(something, who));
+            Ok(())
+        }
+    }
+}
+```
+
+Let's look at the code that shows the definition of `Module` struct after expanded. It has a type parameter `T` which bounds to the module's configuration `Trait`. [PhantomData](https://doc.rust-lang.org/beta/std/marker/struct.PhantomData.html) is used to indicate what type of data our `Module` struct is "tied" to, but not found in our struct.
+
+```rust
+#[cfg(feature = "std")]
+#[structural_match]
+#[rustc_copy_clone_marker]
+pub struct Module<T: Trait>(::std::marker::PhantomData<(T)>);
+```
+
+> *Notes:*
+>
+> * The attribute `#[cfg(feature = "std")]` tells the Rust compiler to only compile our struct when using along with the Rust standard library.
+> * [`structural_match`](https://github.com/rust-lang/rfcs/pull/1445) is used to indicates that instances of the `Module` struct can be used within patterns.
+> * `rustc_copy_clone_marker` indicates the `Module` struct is able to be either copied or cloned.
+
+## Functions and Implementations
+
+### Dispatchable Functions
+
+Dispatchable functions are written by runtime developers and maintain the logic around how to manipulate the blockchain state. Here we only have one function named `do_something` in template runtime module, the expanded code is nearly same with original code except the added `origin` type signature.
+
+```rust
+impl<T: Trait> Module<T> {
+    pub fn do_something(origin: T::Origin, something: u32) -> Result {
+        let who = ensure_signed(origin)?;
+        <Something<T>>::put(something);
+        Self::deposit_event(RawEvent::SomethingStored(something, who));
+        Ok(())
+    }
+}
+```
+
+`Module` struct implements the `Callable` trait in `srml-support` by specifying the associate type to be the module's `Call` enum. Above dispatchable functions will be exposed through this enum, go to its own page for more details about [`Call` enum]((runtime/types/call-enum.md)).
+
+```rust
+impl<T: Trait> ::srml_support::dispatch::Callable for Module<T> {
+    type Call = Call<T>;
+}
+```
+
+There is also a `dispatch` function that redirect the request to the module's `Call` enum:
```

Comment:
This is not so clear, but I think the meat of what we want to teach the user.

If you can expand on this, and the example below, that would be good.

Reply:
Great pickup, I found this function in implementation, but not found where is using it. I will ask other people and loop the result back here.